servlet/src/main/scala/org/http4s/servlet/AsyncHttp4sServlet.scala

```diff
@@ -103,7 +107,13 @@ class AsyncHttp4sServlet[F[_]](
           if (servletRequest.isAsyncStarted)
             servletRequest.getAsyncContext.complete()
         )
-      F.runAsync(f)(loggingAsyncCallback(logger)).unsafeRunSync()
+      val result = F
+        .attempt(f)
+        .flatMap {
+          case Right(()) => F.unit
+          case Left(e) => F.delay(logger.error(e)("Error in error handler"))
+        }
```

Comment:
This should work no?

```suggestion
      val result = F
        .attempt(f)
        .flatMap(loggingAsyncCallback(logger))
```

Reply:
@yanns `loggingAsyncCallback` requires an implicit `Sync`. But we have an implicit `Async` with us.

Reply:
```
trait Async[F[_]] extends AsyncPlatform[F] with Sync[F] with Temporal[F]
```
So it should work no?

Reply:
@yanns I get this error
```
n/scala/org/http4s/servlet/AsyncHttp4sServlet.scala:69:38: could not find implicit value for parameter F: cats.effect.Sync[F] (Could not find an instance of Monad for F)
[error]         .flatMap(loggingAsyncCallback(logger))
```

Reply:
OK thanks for trying.
I don't really understand why, but I'll check later.

Reply:
@yanns I just noticed that the signature of `loggingAsyncCallback` has a return type of `IO[Unit]`. But our dispatcher is a `Dispatcher[F]`. We can summon a `Dispatcher[IO]`. What do you suggest?

Reply:
The return type is `F[Unit]` and not `IO[Unit]` no?

Reply:
@yanns the return type of `loggingAsyncCallback`? It is `IO[Unit]`

```
private[http4s] def loggingAsyncCallback[A](logger: Logger)(
      attempt: Either[Throwable, A]): IO[Unit] =
    attempt match {
      case Left(e) => IO(logger.error(e)("Error in asynchronous callback"))
      case Right(_) => IO.unit
    }
```

Reply:
I don't have the same version as you have:
```
package object internal {

  private[http4s] def loggingAsyncCallback[F[_], A](logger: Logger)(attempt: Either[Throwable, A])(
      implicit F: Sync[F]): F[Unit] =
    attempt match {
      case Left(e) => F.delay(logger.error(e)("Error in asynchronous callback"))
      case Right(_) => F.unit
    }
```

Reply:
Check the version in `main`: https://github.com/http4s/http4s/blob/main/core/src/main/scala/org/http4s/internal/package.scala#L41

Reply:
It was changed by https://github.com/http4s/http4s/commit/4c5ab30eccd09b8c3b88d7941ba0eadd185ec9ce

Reply:
Maybe you should rebase your change on the newest version of `main`?

Reply:
yeah, you were right. I had to rebae with main:) But still this error persists
```
n/scala/org/http4s/servlet/AsyncHttp4sServlet.scala:69:38: could not find implicit value for parameter F: cats.effect.Sync[F] (Could not find an instance of Monad for F)
[error]         .flatMap(loggingAsyncCallback(logger))
```

Reply:
ok thanks for thaving tried!